.github/workflows/run_tests_on_pull_requests.yml

```diff
@@ -0,0 +1,67 @@
+# This workflow will install Python dependencies, run tests,
+# and report test results and code coverage as artifacts. It will only
+# run on pull requests against the main branch.
+
+name: Build and test PR branches
+
+on:
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build_and_test:
```

Comment:
Is there no way to DRY up the run_tests (from publish_docker_image.yml) and this build_and_test below?
I see 7 differences and they all just look like oversights.